lib/seahorn/BvOpsem2WideMemMgr.cc

```diff
@@ -504,10 +503,7 @@ class WideMemManager : public OpSemMemManagerBase {
   PtrTy gep(PtrTy ptr, gep_type_iterator it, gep_type_iterator end) const {
     RawPtrTy rawPtr = m_main.gep(ptr.getRaw(), it, end);
     auto offset = m_main.ptrSub(rawPtr, ptr.getRaw());
-    auto new_size = m_ctx.alu().doSub(
-        ptr.getSize(),
-        m_ctx.alu().doZext(offset, g_slotBitWidth, m_main.wordSzInBits()),
-        g_slotBitWidth);
+    auto new_size = m_ctx.alu().doSub(ptr.getSize(), offset, g_slotBitWidth);
```

Comment:
flagged for efficiency. Symbolically, things will be easier if offset is computed first and then added to both raw part and subtracted from size. Currently, the expressions that are generated will be something like:
```
rawPtr = ptr + offset
newSize = size - (ptr - rawPtr)
```
There is no automatic simplification. So even if size was numeric and offset is numeric, the new size is not numeric.